rl4co/envs/routing/cvrp.py

```diff
@@ -236,7 +240,7 @@ def generate_data(self, batch_size) -> TensorDict:
             {
                 "locs": locs_with_depot[..., 1:, :],
                 "depot": locs_with_depot[..., 0, :],
-                "demand": demand / CAPACITIES[self.num_loc],
+                "demand": demand / self.capacity,
```

Comment:
Great!

Then I think we should add as a check [here](https://github.com/ai4co/rl4co/blob/22497e468aee1de35e89bd0b04dcf4ea7fe71285/rl4co/envs/routing/cvrp.py#L78C1-L79C1) as well - if capacity is not found from the dictionary, then we should set it to 1 and maybe raise a warning, kinda like:

``` python
self.capacity = CAPACITIES.get(num_loc, None) if capacity is None else capacity
 if self.capacity is None:
    self.capacity = 1
    log.warn("Capacity not found in dictionary. Please pass a value manually when instantiating the class")
```

Reply:
Otherwise, we would divide by `None`

Reply:
We already throw an error when initialising using the (parent class’) _init function, though, do we still need another check here? I.e. can we assume that the capacity could go lost along the way?

Reply:
My bad, you're right ;)